vortex-expr/src/pruning/mod.rs

```diff
@@ -1,295 +1,39 @@
-        let other_col = lit(42);
-        let not_eq_expr = gt(get_item_scope(column.clone()), other_col.clone());
-
-        let (converted, refs) = convert_to_pruning_expression(&not_eq_expr);
-        assert_eq!(
-            refs.map(),
-            &HashMap::from_iter([(
-                FieldOrIdentity::Field(column.clone()),
-                HashSet::from_iter([Stat::Max])
-            ),])
-        );
-        let expected_expr = lt_eq(
-            get_item_scope(stat_field_name(&column, Stat::Max)),
-            other_col.clone(),
-        );
-        assert_eq!(&converted, &expected_expr);
-    }
-
-    #[test]
-    pub fn pruning_lt_column() {
-        let column = FieldName::from("a");
-        let other_col = FieldName::from("b");
-        let other_expr = get_item_scope(other_col.clone());
-        let not_eq_expr = lt(get_item_scope(column.clone()), other_expr.clone());
-
-        let (converted, refs) = convert_to_pruning_expression(&not_eq_expr);
-        assert_eq!(
-            refs.map(),
-            &HashMap::from_iter([
-                (
-                    FieldOrIdentity::Field(column.clone()),
-                    HashSet::from_iter([Stat::Min])
-                ),
-                (
-                    FieldOrIdentity::Field(other_col.clone()),
-                    HashSet::from_iter([Stat::Max])
-                )
-            ])
-        );
-        let expected_expr = gt_eq(
-            get_item_scope(stat_field_name(&column, Stat::Min)),
-            get_item_scope(stat_field_name(&other_col, Stat::Max)),
-        );
-        assert_eq!(&converted, &expected_expr);
-    }
-
-    #[test]
-    pub fn pruning_lt_value() {
-        let column = FieldName::from("a");
-        let other_col = lit(42);
-        let not_eq_expr = lt(get_item_scope(column.clone()), other_col.clone());
-
-        let (converted, refs) = convert_to_pruning_expression(&not_eq_expr);
-        assert_eq!(
-            refs.map(),
-            &HashMap::from_iter([(
-                FieldOrIdentity::Field(column.clone()),
-                HashSet::from_iter([Stat::Min])
-            )])
-        );
-        let expected_expr = gt_eq(
-            get_item_scope(stat_field_name(&column, Stat::Min)),
-            other_col.clone(),
-        );
-        assert_eq!(&converted, &expected_expr);
+pub trait StatsCatalog {
+    // Given an id, field and stat return an expression that when evaluated will return that stat
+    // this would be a column reference or a literal value, if the value is known at planning time.
+    // TODO(joe): replace field with a field path, once implemented.
+    fn stats_ref(&mut self, _id: &Identifier, _field: &FieldPath, _stat: Stat) -> Option<ExprRef> {
+        None
     }
+}
 
-    #[test]
-    fn unprojectable_expr() {
-        let or_expr = not(lt(get_item_scope("a"), get_item_scope("b")));
-        assert!(PruningPredicate::try_new(&or_expr).is_none());
+pub trait AnalysisExpr {
```

Comment:
On the subject of names, should this be `FalsifiableExpr`? or something to that effect?

Reply:
I want this to work for thing other than pruning (like I would even add simplify here

Reply:
I am not sure if we would want to split that into more traits but that add the complexity of adding a new expr